libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -707,6 +707,7 @@ private void shutdown() {
     navigationViewEventDispatcher.onDestroy(navigationViewModel.retrieveNavigation());
     mapView.onDestroy();
     navigationViewModel.onDestroy(isChangingConfigurations());
+    ((FragmentActivity) getContext()).getViewModelStore().clear();
```

Comment:
wonder if this view is being used with `Activity` would this call result in `ClassCastException`?